test/index.js

```diff
@@ -616,8 +616,9 @@ describe('def', function() {
 
     throws(function() { defaultTo(0, 'XXX'); },
            errorEq(TypeError,
-                   '‘defaultTo’ expected a value of type (Nullable Number) ' +
-                   'as its second argument; received "XXX"'));
+                   '‘defaultTo’ expected its first and second arguments to ' +
+                   'be of the same type; the first is of type Number' +
+                   ', however the second, "XXX", is of type String'));
```

Comment:
If I understand correctly, this is what I meant by

> the error messages could be improved for unary and binary types, I wasn't sure exactly how to word it though

Do you have any thoughts on how to say this nicely? Maybe something like

`‘defaultTo’ expected its first and second arguments to ??? (derive from the same type? be of the same base type?); the first is of type Number, so the second was expected to be of type (Nullable Number) but received "XXX", which is of type String`